rest/serverless_test.go

```diff
@@ -579,3 +579,43 @@ func TestServerlessUnsuspendAdminAuth(t *testing.T) {
 	resp = rt.SendAdminRequestWithAuth(http.MethodGet, "/invaliddb/doc", "", base.TestClusterUsername(), base.TestClusterPassword())
 	assertHTTPErrorReason(t, resp, http.StatusForbidden, "")
 }
+
+func TestImportPartitionsServerless(t *testing.T) {
+	tests := []struct {
+		name               string
+		importPartition    *uint16
+		expectedPartitions *uint16
+	}{
+		{
+			name:               "serverless partitions",
+			expectedPartitions: base.Uint16Ptr(6),
+		},
+		{
+			name:               "serverless partitions with import_partition specified",
+			importPartition:    base.Uint16Ptr(8),
+			expectedPartitions: base.Uint16Ptr(8),
+		},
```

Comment:
Is it worth adding the test case for `serverless: false` == `16` or do we have sufficient coverage elsewhere?

Reply:
Good point! Updated the PR to add this test case in.